docs/rfcs/env-vars.md

```diff
@@ -0,0 +1,216 @@
+resolution (e.g. `${env:${http://example.com}}` would get the
+environment variable whose name is stored in the URL
+`http://example.com`).
+
+The environment variable value is parsed by the yaml.v3 parser to an
+any-typed variable. The yaml.v3 parser mostly follows the YAML v1.2
+specification with [*some exceptions*](https://github.com/go-yaml/yaml#compatibility). 
+You can see
+how it works for some edge cases in this example:
+[`go.dev/play/p/RtPmH8aZA1X`](https://go.dev/play/p/RtPmH8aZA1X).
+
+### Issues of current behavior
+
+#### Unintuitive behavior on unset environment variables
+
+When an environment variable is empty, all syntaxes return an empty
+string with no warning given; this is frequently unexpected but can also
+be used intentionally. This is especially unintuitive when the user did
+not expect expansion to happen. Two examples where this is unexpected
+are the following:
+
+1.  **Opaque values such as passwords that contain `$`** (issue
+    [#8215](https://github.com/open-telemetry/opentelemetry-collector/issues/8215)).
+    If the $ is followed by an alphanumeric character or one of the
+    special characters, it's going to lead to false positives.
+2.  **Prometheus relabel config** (issue
+    [`contrib#9984`](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9984)).
+    Prometheus uses `${1}` in some of its configuration values. We
+    resolve this to the value of the environment variable with name
+    '`1`'.
+3.  **Other uses of $** (issue
+    [`contrib#11846`](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11846)).
+    If a product requires the use of `$` in some field, we would most
+    likely interpret it as an environment variable. This is not
+    intuitive for users.
+
+#### Unexpected type casting
+
+When using the env syntax we parse its value as YAML. Even if you are
+familiar with YAML, because of the implicit type casting rules and the
+way we store intermediate values, we can get unintuitive results.
+
+The most clear example of this is issue
+[*#8565*](https://github.com/open-telemetry/opentelemetry-collector/issues/8565):
+When setting a variable to value `0123` and using it in a string-typed
+field, it will end up as the string `"83"` (where as the user would
+expect the string to be `0123`).
+
+#### We are less restrictive than the Configuration WG
+
+The Configuration WG defines an [*environment variable expansion feature
+for SDK
+configurations*](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/file-configuration.md#environment-variable-substitution).
+This accepts only non empty alphanumeric + underscore identifiers
+starting with alphabetic or underscore. If the Configuration WG were to
+expand this in the future (e.g. to include other features present in
+Bash-like syntax), we would not be able to expand our braces syntax to
+support new features without breaking users.
```

Comment:
As an example, open-telemetry/opentelemetry-specification/pull/3948 proposes to add the Bash syntax for specifying a default